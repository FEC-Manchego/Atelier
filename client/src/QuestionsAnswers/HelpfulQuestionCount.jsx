import React from 'react';
import $ from 'jquery';

const HelpfulQuestionCount = (props) => {
  const helpfulQuestionAjax = () => {
    $.ajax({
      method: 'POST',
      url: 'http://localhost:3000/QuestionHelpful',
      contentType: 'application/json',
      success: () => {
        props.ajaxGetQuestionHelpful();
        console.log('success question counter++');
      },
      error: (err) => {
        console.log(err);
      },
    });
  };

  return (
    <div className='helpfulQuestionCounter'>
      Helpful?
      <button type="submit" onClick={helpfulQuestionAjax} className="reporthelpful-btn">Yes</button>
      {/* should be the helpfulcount state from questionsanswerbox file but not workinging */}
      {props.questionhelpfulCount}
    </div>
  );
};

export default HelpfulQuestionCount;
