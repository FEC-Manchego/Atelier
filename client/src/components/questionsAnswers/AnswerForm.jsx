import React from 'react';
import $ from 'jquery';
import config from '../../../../config.js';

class AnswerForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      showForm: false,
      currentQID: props.qid,
      currentPID: props.pid,
      body: '',
      bold: false,
      name: '',
      email: '',
      photo: [],
      selectedFile: '',
      nameValid: true,
      emailValid: true,
      bodyValid: true
    };

    this.onClick = this.onClick.bind(this);
    this.onChangeBody = this.onChangeBody.bind(this);
    this.onChangeName = this.onChangeName.bind(this);
    this.onChangeEmail = this.onChangeEmail.bind(this);
    this.showForm = this.showForm.bind(this);
    this.addAnswer = this.addAnswer.bind(this);
    this.onCloseForm = this.onCloseForm.bind(this);
    this.fileSelectorHandler = this.fileSelectorHandler.bind(this);
    this.fileUploaderHandler = this.fileUploaderHandler.bind(this);
    this.formValidation = this.formValidation.bind(this);
  }

  fileSelectorHandler (e) {
    this.setState({
      selectedFile: e.target.files[0]
    });
  }

  fileUploaderHandler(e) {
    e.preventDefault();
    var form = new FormData();
    form.append("image", this.state.selectedFile);
    var options = {
      "url": `https://api.imgbb.com/1/upload?key=${config.imgBBTokeb}`,
      "method": "POST",
      "timeout": 0,
      "processData": false,
      "mimeType": "multipart/form-data",
      "contentType": false,
      "data": form
    };

    $.ajax(options).done((response) => {
      var res = JSON.parse(response);
      if (this.state.photo.length <= 5) {
        this.setState({
          photo: [...this.state.photo, res.data.url]
        });
      } else {
        alert('5 photos max');
      }
    });
  }

  onClick() {
    this.setState({ showForm: true });
  }

  onChangeBody(e) {
    this.setState({
      body: e.target.value,
    });
  }

  onCloseForm() {
    if (this.state.showForm === true) {
      console.log('clciked');
      this.setState({
        showForm: false,
      });
    }
  }

  onChangeName(e) {
    if (e.target.value.toLowerCase() === 'seller') {
      this.setState({
        name: e.target.value,
        bold: true,
      });
    } else {
      this.setState({
        name: e.target.value,
      });
    }
  }

  onChangeEmail(e) {
    this.setState({
      email: e.target.value,
    });
  }

  formValidation() {
    const name = this.state.name;
    const body = this.state.body;
    const email = this.state.email;

    let validForm = true;

    if (!name) {
      validForm = false;
      this.setState({
        nameValid: false
      });
    }

    if (!body) {
      validForm = false;
      this.setState({
        bodyValid: false
      });
    }

    if (!email) {
      validForm = false;
      this.setState({
        emailValid: false
      });
    }

    if (name === 'undefined') {
      if (!name.match(/^[a-zA-Z0-9]+$/)) {
        validForm = false;
        this.setState({
          nameValid: false
        });
      }
    }

    if ( name && body && email) {
      return validForm;
    }
  }

  addAnswer(e) {
    e.preventDefault();

    if (this.formValidation()) {
      const info = {
        body: this.state.body,
        name: this.state.name,
        email: this.state.email,
        photos: this.state.photo,
        question_id: this.state.currentQID
      };

      $.ajax({
        method: 'POST',
        url: 'http://localhost:3000/addAnswer',
        contentType: 'application/json',
        data: JSON.stringify(info),
        success: () => {
          this.setState({
            body: '',
            name: '',
            email: '',
            photos: [],
            showForm: false,
          });
          console.log('add answer success post');
        },
        error: () => {
          console.log('error in addAnswers');
        },
      });
    } else {
      alert('NOT VALID FORM');
    }
    }


  showForm() {
    const divStyle = {
      margin: '0px',
      height: '200px',
      width: '279px',
      fontSize: '1em'
    };
    const text = {
      fontSize: '1em'
    };

    return (
      <div className="aboxcenter">
        <form>
          <button type="submit" className='X' onClick={this.onCloseForm}>X</button>
          <h1 className='answerboxtitle'>Add Answer</h1>

          <label className="field field_v1">
            <input
              className="field__input"
              value={this.state.name}
              onChange={this.onChangeName}
              type="text"
              name="name"
              maxLength="60"
              placeholder="Example: jack543!**"
            />
            <span className="field__label-wrap">
              <span className="field__label">Enter UserName*</span>
            </span>
          </label>
          < br/>

          < br/>
          <label style={text}> For privacy reasons, do not use your full name or email address**</label>
          < br/>


          < br/>
          <label className="field field_v1">
            <input
              className="field__input"
              value={this.state.email}
              onChange={this.onChangeEmail}
              type="text"
              name="Email"
              placeholder="Example: jack@gmail.com*"
              pattern="/^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/"
            />
            <span className="field__label-wrap">
              <span className="field__label">Enter Email Here*</span>
            </span>
          </label>
          < br/>

          < br/>
          <label style={text}> For authentication reasons, you will not be emailed**</label>
          < br/>


          < br/>
          <label className="field field_v1">
            <textarea
              className="field__input"
              value={this.state.body}
              onChange={this.onChangeBody}
              type="text"
              name="body"
              cols="50"
              rows="50"
              style={divStyle}
              placeholder="Example: Is this pretty!**"
            />
            <span className="field__label-wrap">
              <span className="field__label center"> Enter Answer Here*</span>
            </span>
          </label>

          < br/>

          <input className='answerbutton upload'
            type="submit"
            onClick={this.addAnswer}
            value="Post Answer"
          />

          <div className="upload">
            <div className="">
              <input type="file" onChange={this.fileSelectorHandler} accept="image/*" />
              <button onClick={this.fileUploaderHandler}> Upload Photo</button>
              < br/>

              {this.state.photo.map((photo, index) => {
                return (
                  <div className='uploadimg img-spacing' key={index}>
                    <img src={photo} className='uploadimg' />
                  </div>
                );
              })}
            </div>
          </div>
        </form>
      </div>
    );
  }

  render() {
    const { showForm } = this.state;
    return (
      <div>
        <button className="addanswer" type="submit" onClick={ this.onClick }>
          Add Answer
          {showForm && this.showForm()}
        </button>
      </div>
    );
  }
}

export default AnswerForm;