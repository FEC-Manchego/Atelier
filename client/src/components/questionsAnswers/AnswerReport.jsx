import React from 'react';

class AnswerReport extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      reported: false,
      reportedArray: [],
      reportedID: this.aID = this.props,
    };
    this.reportedBtn = this.reportedBtn.bind(this);
  }

  reportedBtn() {
    if (!this.state.reportedArray.includes(this.state.reportedID)) {
      this.setState({
        reported: true,
      });
    } else {
      this.setState({
        reported: false,
      });
    }
  }

  render() {
    return (
      <div>
        <button type="submit" onClick={this.reportedBtn} className="reporthelpful-btn">
          {this.state.reported ? 'Reported' : 'Report'}
        </button>
      </div>
    );
  }
}


export default AnswerReport;
