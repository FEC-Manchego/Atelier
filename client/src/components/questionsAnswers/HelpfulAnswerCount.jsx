import React from 'react';
import $ from 'jquery';

class HelpfulAnswerCount extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      ahelpid: props.id,
      pid: props.mainProductId,
      ahelpfulcount: props.defaultcounter,
    };

    this.helpfulAincrementer = this.helpfulAincrementer.bind(this);
    this.increasecounter = this.increasecounter.bind(this);
  }

  increasecounter() {
    this.setState({
      ahelpfulcount: this.state.ahelpfulcount + 1,
    });
  }

  helpfulAincrementer() {
    $.ajax({
      method: 'POST',
      url: `http://localhost:3000/ahelpful`,
      contentType: 'application/json',
      data: JSON.stringify({ ahelpid: this.state.ahelpid }),
      success: () => {
        this.increasecounter();
        console.log('helpfulAincrementer++');
      },
      error: () => {
        console.log('err helpfulAnswerAjax');
      },
    });
  };

  render() {
    return (
      <div className='helpfulQuestionCounter'>
        HelpFul? {this.state.ahelpfulcount} _
        <button type="submit" onClick={this.helpfulAincrementer} className="answerhelpfulbtn">Yes</button>
      </div>
    );
  };
}

export default HelpfulAnswerCount;






