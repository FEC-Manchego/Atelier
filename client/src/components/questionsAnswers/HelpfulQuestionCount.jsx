import React from 'react';
import $ from 'jquery';

const HelpfulQuestionCount = () => {
  const helpfulQincrementer = () => {
    $.ajax({
      method: 'POST',
      url: 'http://localhost:3000/qhelpful',
      contentType: 'application/json',
      success: () => {
        console.log('helpfulQincrementer++ ');
      },
      error: () => {
        console.log('err helpfulAnswerAjax');
      },
    })
    // .then (() => {
    //   $.ajax({
    //     method: 'GET',
    //     url: 'http://localhost:3000/qhelpful',
    //     contentType: 'application/json',
    //     success: () => {
    //       console.log('helpfulQincrementer++ ');
    //     },
    //     error: () => {
    //       console.log('err helpfulAnswerAjax');
    //     },
    //   })
    // })
  };

  return (
    <div className="helpfulQuestionCounter">
      Question Helpful?
      <button type="submit" onClick={helpfulQincrementer} className="questionhelpfulbtn">Yes</button>
      {/* should be the helpfulcount state from questionsanswerbox file but not workinging */}
      {/* {test} */}
    </div>
  );
};

export default HelpfulQuestionCount;

// fetch('http://localhost:3000/question')
// .then((response) => response.json())
// .then((helpfulCounter) => {
//    test = helpfulCounter.results.map( data => {
//     return data.question_helpfulness
//    })
//    console.log('this is question count', test)

// });