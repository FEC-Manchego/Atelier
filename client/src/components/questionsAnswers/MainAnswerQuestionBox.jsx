import React from 'react';
import Moment from 'moment';
import AskQuestions from './AskQuestionButtons.jsx';
import AnswerReport from './AnswerReport.jsx';
import QuestionReport from './QuestionReport.jsx';
import HelpfulAnswerCount from './HelpfulAnswerCount.jsx';
import HelpfulQuestionCount from './HelpfulQuestionCount.jsx';
import AnswerForm from './AnswerForm.jsx';

const MainAnswerQuestionBox = (props) => {
  const QABOX = props.questionanswerslist.slice(0, props.visibleQuestions).map((data, index) => (
    <div key={data.qID}>
      <div className="questions-asked">
        <ul>
          <span className="questiontext">
            <b>
              Q:
            </b>
            { data.question }
            <div className="HelpfulQuestionCount">
              <HelpfulQuestionCount id={data.qID} />
              _
              {data.questionHelpful}
              _
              <p className="questionhelpfont">{ data.question_helpfulness }</p>
              <QuestionReport qid={data.qID} />
              <AnswerForm qid={data.qID} />
            </div>
          </span>
        </ul>
      </div>

      <div className="questions-asked">
        <div>
          <ul>
            {data.answers.slice(0, props.visibleAnswers).map(answerlist => (
              <div key={answerlist.id}>
              <p className='indent'> A: {answerlist.body}
                <br/>
                <br/>
                <p className="user-data">by: {answerlist.answerer_name}
                  __
                  <Moment format="MMMM-DD-YYYY" date={answerlist.date} />
                  __
                  HelpFul?
                  <HelpfulAnswerCount id={answerlist.id}/> _{answerlist.helpfulness}
                  __
                  <AnswerReport aID={answerlist.id} />
                </p>
              </p>
              </div>
            ))}
          </ul>
          </div>

        {/* <div className="answerPhotos">
          <AnswerPhotos answersList={props.answersList} />
        </div> */}
      </div>
    </div>

  ));

  return (
    <div className="container">
      {QABOX}
      <AskQuestions
        loadMoreAnswers={props.loadMoreAnswers}
        loadMoreQuestions={props.loadMoreQuestions}
        btnvisible={props.btnvisible}
        btnvisibleq={props.btnvisibleq}
        mainProductId={props.mainProductId}
      />
    </div>

  );
};

export default MainAnswerQuestionBox;
