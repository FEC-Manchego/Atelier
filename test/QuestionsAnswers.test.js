import React from 'react';

import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import AnswerForm from '../client/src/components/questionsAnswers/AnswerForm.jsx';
//import AnswerPhotos from '../client/src/components/questionsAnswers/AnswerPhotos.jsx.jsx';
import AnswerReport from '../client/src/components/questionsAnswers/AnswerReport.jsx';
import AskQuestions from '../client/src/components/questionsAnswers/AskQuestionButtons.jsx'
import MainAnswerQuestionBox from '../client/src/components/questionsAnswers/MainAnswerQuestionBox.jsx'
import QuestionsAnswersState from '../client/src/components/questionsAnswers/QuestionAnswerState.jsx'
import HelpfulAnswerCount from '../client/src/components/questionsAnswers/HelpfulAnswerCount.jsx'
import HelpfulQuestionCount from '../client/src/components/questionsAnswers/HelpfulQuestionCount.jsx'

Enzyme.configure({ adapter: new Adapter() });

//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing Answer Form
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»

describe('AnswerForm', () => {
  it('should be true that AnswerForm exists', () => {
    const wrapper = shallow(<AnswerForm />);
    expect(wrapper.find(AnswerForm)).toBeTruthy();
  });
});

describe('AnswerForm', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<AnswerForm />);
    expect(wrapper).toMatchSnapshot();
  });
});

// Testing AnswerPhotos Form
// describe('AnswerPhotos', () => {
//   it('renders correctly', () => {
//     const wrapper = shallow(<AnswerPhotos />);
//     expect(wrapper).toMatchSnapshot();
//     // On the first run of this test, Jest will generate a snapshot file automatically.
//   });
// });


//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing Answer Report
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»

describe('AnswerReport', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<AnswerReport />);
    expect(wrapper).toMatchSnapshot();
  });
});


//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing AskQuestionButtons Form
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»

describe('AskQuestions', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<AskQuestions />);
    expect(wrapper).toMatchSnapshot();
  });
});


//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing HelpfulAnswersCount Form
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»

describe('HelpfulAnswerCount', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<HelpfulAnswerCount />);
    expect(wrapper).toMatchSnapshot();
  });
});


//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing HelpfulQuestionCount Form
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»

describe('HelpfulQuestionCount', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<HelpfulQuestionCount />);
    expect(wrapper).toMatchSnapshot();
  });
});


//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing LoadMoreAnswers Form
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»



//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing MainAnswersQuestionBox Form
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»

describe('MainAnswerQuestionBox', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<MainAnswerQuestionBox />);
    expect(wrapper).toMatchSnapshot();
  });
});


//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing QuestionAnswerState Form
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»

describe('QuestionsAnswersState', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<QuestionsAnswersState />);
    expect(wrapper).toMatchSnapshot();
  });
});




//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing QuestionForm Form
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»




//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing QuestionReport Form
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»




//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»
// Testing SearchBar Form
//ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»ğŸ‘»

