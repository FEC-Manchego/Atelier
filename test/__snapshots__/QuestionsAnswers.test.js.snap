// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerForm renders correctly 1`] = `ShallowWrapper {}`;

exports[`AnswerReport renders correctly 1`] = `ShallowWrapper {}`;

exports[`AskQuestions renders correctly 1`] = `ShallowWrapper {}`;

exports[`HelpfulAnswerCount renders correctly 1`] = `ShallowWrapper {}`;

exports[`QuestionsAnswersState renders correctly 1`] = `ShallowWrapper {}`;
